### PR TITLE
feature(Payroll for HGR)

### LINKED ISSUE #7158: Implementation of Payroll for General Hospitals of Reference
After completing a mission in the 3 HGR of Tshikapa city, we noticed that the Payroll system differs slightly from the one implemented for the Baptist community hospitals. To address this, we considered using the BHIMA system, which can be configured to accommodate the necessary changes. However, implementing this would require performing certain tasks manually to ensure accuracy and prevent errors. Our proposal is to integrate these additional functionalities into the BHIMA system.

After conducting a thorough analysis of the current situation, we have identified that the BHIMA system currently only handles grades and functions, without consolidating functions and professions together. This poses a potential challenge and may lead to confusion. To address this, we propose enhancing the system by incorporating the management of professions. This will help avoid any potential confusion and ensure a more streamlined and efficient process.

### PR DESCRIPTION
**Pull request by lomamech**
- Separation between Job title and function
- Updated employee management by adding support for job title management
- Improved Payroll index module to support general reference hospitals model

closes #7158

### CODE REVIEW

File: server/models/schema.sql [hunk 1]
```diff
@@ -524,6 +524,7 @@ CREATE TABLE `employee` (
   `patient_uuid`  BINARY(16) DEFAULT NULL,
   `is_medical`    TINYINT(1) DEFAULT 0,
```
Comment (jniles) [inline server/models/schema.sql:525]: Can we move the `is_medical` to the `title_employee` table?  It seems like `is_medical` has to do with the employee's job title, not the employee themselves, right?

From this implementation, I assume all employees have titles?  Can multiple employees have the same title?

Comment (lomamech) [reply server/models/schema.sql:525]: That's a really good idea that would make things a lot easier. I can do it in another PR

**Comment by lomamech** [comment]
[PAYROLL FOR HGR.pdf](https://github.com/IMA-WorldHealth/bhima/files/12157578/PAYROLL.FOR.HGR.pdf)
[PAYROLL POUR LE HGR AVEC INDICE.pdf](https://github.com/IMA-WorldHealth/bhima/files/12157579/PAYROLL.POUR.LE.HGR.AVEC.INDICE.pdf)

Here are sample SQL data files (source these after loading bhima_test by doing `build:db`):
  - [payrollSetup.sql.txt](https://github.com/IMA-WorldHealth/bhima/files/12369959/payrollSetup.sql.txt)
  - [employeeSample.sql.txt](https://github.com/IMA-WorldHealth/bhima/files/12369963/employeeSample.sql.txt)

**Comment by lomamech** [comment]
[Uploading MODELE PAYROLL HGR.xlsx…]()

**Comment by lomamech** [comment]
[MODELE PAYROLL HGR.xlsx](https://github.com/IMA-WorldHealth/bhima/files/12182368/MODELE.PAYROLL.HGR.xlsx)

**Review by jmcameron — APPROVED** [review]
I did a brief look through the code and it seems okay.
I've tested this PR (with training and assistance from Chris) and it seems to work fine.

I suggest we go ahead and merge it.  I have some improvements to do (eg, for some of the English translations) but I plan to do those later.

**Pull request merged**